pkg/reconciler/vertex/scaling/scaling.go

```diff
@@ -223,6 +215,37 @@ func (s *Scaler) scaleOneVertex(ctx context.Context, key string, worker int) err
 		s.daemonClientsCache.Add(pl.GetDaemonServiceURL(), daemonClient)
 	}
 
+	partitionBufferLengths, partitionAvailableBufferLengths, totalBufferLength, totalCurrentPending, err := getBufferInfos(ctx, key, daemonClient, pl, vertex)
+	if err != nil {
+		err := fmt.Errorf("error while fetching buffer info, %w", err)
+		return err
+	}
```

Comment:
@akashjkhamkar , before this change, `getBufferInfos` was only invoked when the vertex is NOT a source. Now it gets invoked when the vertex is a source. I wonder if that's the reason why we transiently not able to create the source pod. Could you verify?

Reply:
`getBufferInfos ` will only be effective, if the vertex has owned buffers (which is true only for non-source)

so i dont think it has to do with getBufferInfos